### PR TITLE
[Project] Add pipeline runs deletion to project deletion cascade flow

### PR DESCRIPTION
**Pull request by Tankilevitch**
Adding deletion of pipeline runs as part of project deletion so we don't leave any leftovers of project resources

Fixes https://jira.iguazeng.com/browse/ML-4119

### CODE REVIEW

File: mlrun/api/crud/pipelines.py [hunk 1]
```diff
@@ -102,6 +98,38 @@ def list_pipelines(
...
+            project=project,
+            format_=mlrun.common.schemas.PipelinesFormat.metadata_only,
+        )
+        kfp_client = self.initialize_kfp_client()
+
+        if project_pipeline_runs:
+            logger.debug(
+                "Detected pipeline runs for project, deleting them",
+                project=project,
+                pipeline_runs=[run["id"] for run in project_pipeline_runs],
```
Comment (TomerShor) [inline mlrun/api/crud/pipelines.py:113]:
```suggestion
                pipeline_run_ids=[run["id"] for run in project_pipeline_runs],
```

File: mlrun/api/crud/pipelines.py [hunk 2]
```diff
@@ -102,6 +98,38 @@ def list_pipelines(
...
+
+        for pipeline_run in project_pipeline_runs:
+            try:
+                # delete pipeline run also terminates it if it is in progress
+                kfp_client._run_api.delete_run(pipeline_run["id"])
+            except Exception as exc:
+                # we don't want to fail the entire delete operation if we failed to delete a single pipeline run
+                # (we will log the error and continue) because it is being used as part of the delete project
+                # operation and we don't want to fail the entire delete project operation because of a single
+                # pipeline run
```
Comment (TomerShor) [inline mlrun/api/crud/pipelines.py:124]: No need to write it twice
```suggestion
                # we don't want to fail the entire delete operation if we failed to delete a single pipeline run
                # so it won't fail the delete project operation. we will log the error and continue
```

File: mlrun/api/crud/pipelines.py [hunk 3]
```diff
@@ -102,6 +98,38 @@ def list_pipelines(
...
+                logger.warning(
+                    "Failed to delete pipeline run",
+                    project=project,
+                    pipeline_run=pipeline_run,
```
Comment (TomerShor) [inline mlrun/api/crud/pipelines.py:128]: is the entire run required in the log? maybe just the `id ` so won't be spammy

**Review by TomerShor — CHANGES_REQUESTED** [review]
Minor stuff

File: mlrun/api/crud/pipelines.py (hunk 3 above)
Comment (liranbg) [reply mlrun/api/crud/pipelines.py:128]: also project, need to log only its name

Comment (Tankilevitch) [reply mlrun/api/crud/pipelines.py:128]: project is already of type `str`

**Review by TomerShor — APPROVED** [review]
🚀

File: mlrun/api/crud/pipelines.py [hunk 4]
```diff
@@ -102,6 +98,36 @@ def list_pipelines(
...
+                # delete pipeline run also terminates it if it is in progress
+                kfp_client._run_api.delete_run(pipeline_run["id"])
+            except Exception as exc:
+                # we don't want to fail the entire delete operation if we failed to delete a single pipeline run
+                # so it won't fail the delete project operation. we will log the error and continue
+                logger.warning(
+                    "Failed to delete pipeline run",
+                    project=project,
+                    pipeline_run_id=pipeline_run["id"],
+                    exc_info=exc,
```
Comment (liranbg) [inline mlrun/api/crud/pipelines.py:127]:
```suggestion
                    exc_info=err_to_str(exc),
```

**Review by liranbg — COMMENTED** [review]
minor :)

File: mlrun/api/crud/pipelines.py (hunk 4 above)
Comment (liranbg) [reply mlrun/api/crud/pipelines.py:127]: talked offline, exc_info format within logger itself

**Review by liranbg — APPROVED** [review]

**Pull request merged**